### PR TITLE
split network utilities into their own crates

### PR DESCRIPTION
**Pull request by divagant-martian**
## Change checklist

- [ ] Self-review.
- [ ] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [ ] Tests if relevant.
- [ ] All breaking changes documented.

### CODE REVIEW

File: net-tools/net/Cargo.toml
Comment (flub) [inline net-tools/net/Cargo.toml:1]: I imagine we'll further split this up broadly as:
- interfaces & LocalAddresses (which is a view of iterfaces really)
- udp socket

And then somehow figure out where the remaining small bits go.

But leaving this as one "net" is fine for now I think.  Small steps are easier.

Comment (divagant-martian) [reply net-tools/net/Cargo.toml:1]: yeah it's not great, and the name is not great either, but it's the best way to progress on this for the time being

Comment (dignifiedquire) [reply net-tools/net/Cargo.toml:1]: we will need sth else for next week, otherwise we can’t release it (I don’t think net is free on crates.io)

**Comment by dignifiedquire** [comment]
@divagant-martian can this be closedß

**Pull request closed without merging**